viewer/packages/data-providers/src/image-360-data-providers/descriptor-providers/datamodels/cdm/Cdf360CdmBatchCollectionLoader.ts

```diff
@@ -0,0 +1,424 @@
+    this._cogniteSdk = cogniteSdk;
+    this._batchSize = BATCH_SIZE;
+    this._batchDelay = BATCH_DELAY_MS;
+  }
+
+  /**
+   * Request descriptors for a collection. This will be batched with other concurrent requests.
+   * Batches are processed sequentially to avoid overwhelming the API.
+   */
+  public async getCollectionDescriptors(identifier: {
+    externalId: string;
+    space: string;
+  }): Promise<Historical360ImageSet<DMDataSourceType>[]> {
+    return new Promise((resolve, reject) => {
+      this._pendingBatch.push({ identifier, resolve, reject });
+
+      // Clear existing timer
+      if (this._batchTimer) {
+        clearTimeout(this._batchTimer);
+      }
+
+      // If we've hit the batch size, execute immediately (if not already processing)
+      if (this._pendingBatch.length >= this._batchSize) {
+        void this.tryExecuteBatch();
+      } else {
+        // Otherwise, wait a bit to accumulate more requests
+        this._batchTimer = setTimeout(() => {
+          void this.tryExecuteBatch();
+        }, this._batchDelay);
+      }
+    });
+  }
+
+  /**
+   * Try to execute a batch. If already processing, this will be a no-op.
+   * The next batch will be picked up when the current one finishes.
+   */
+  private async tryExecuteBatch(): Promise<void> {
+    if (this._isProcessing || this._pendingBatch.length === 0) {
+      return;
+    }
+
+    this._isProcessing = true;
+    try {
+      await this.executeBatch();
+    } finally {
+      this._isProcessing = false;
+
+      // If there are more pending requests, process the next batch
+      if (this._pendingBatch.length > 0) {
+        // Small delay before next batch to avoid rapid-fire requests
+        setTimeout(() => {
+          void this.tryExecuteBatch();
+        }, 50);
+      }
+    }
+  }
+
+  private async executeBatch(): Promise<void> {
+    if (this._pendingBatch.length === 0) return;
+
+    // Take current batch and reset for next batch
+    const batch = this._pendingBatch;
+    this._pendingBatch = [];
+    this._batchTimer = null;
+
+    const collectionRefs = batch.map(item => item.identifier);
+
+    try {
+      // Execute batch query directly - no throttling needed
+      // The batching itself (BATCH_SIZE queries instead of 1000) prevents API overload
+      const query = get360CdmCollectionsQuery(collectionRefs);
+
+      const result = await this._dmsSdk.queryNodesAndEdges<CdfImage360CollectionDmQuery>(query);
+
+      // Group results by collection
+      const resultsByCollection = await this.groupResultsByCollection(result);
+
+      // Resolve each request with its specific results
+      batch.forEach(({ identifier, resolve }) => {
+        const key = `${identifier.externalId}:${identifier.space}`;
+        const descriptors = resultsByCollection.get(key);
+
+        if (descriptors) {
+          resolve(descriptors);
+        } else {
+          // Collection not found
+          console.warn(`Collection not found: ${key}`);
```

Comment:
Could we add more info here to make it easier to debug if it triggers? E.g. `` console.warn(`[${Cdf360CdmBatchCollectionLoader.name}]: ...`); ``

Same for the rest of the `console.warn/error`s in this file